tests/common.go

```diff
@@ -127,16 +128,25 @@ func ValidateCleanup() {
 }
 
 // ValidateContext is the ginkgo spec for validating a scheduled context
-func ValidateContext(ctx *scheduler.Context) {
+func ValidateContext(ctx *scheduler.Context, vState int) {
 	ginkgo.Describe(fmt.Sprintf("For validation of %s app", ctx.App.Key), func() {
 		Step(fmt.Sprintf("validate %s app's volumes", ctx.App.Key), func() {
-			ValidateVolumes(ctx)
+			ValidateVolumes(ctx, vState)
```

Comment:
are we changing vState somewhere? it seems it might be a problem if we're using copy instead of reference

Reply:
vState stands for VolumeState, in a test a volume is expected to be in pending state. The test will pass this flag to inform the ValidateVolumes to pass the  test  even if  the volume is in pending state

Reply:
I don't think that changing `ValidateContext()` like this should be done.
It is not obvious what the `vState` variable is, and what it's impact on the function is.

You are passing in defaultInterval which is a context from many places into this function.  You essentially have two functions:

One where defaultInterval is 0,
another where defaultInterval is 1.

I suggest that you create two functions:


 * `ValidateContextXXX1()`
 * `ValidateContextXXX2()`

replace `XXX1` and `XXX2` with obvious names as to how the functions are behaving.

Then replace ValidateContext() with ValidateConextXXX?() depending on what you need.

Reply:
@rodrigc , creating 2 functions might create problem in case there is a future development in validateVolume and the changes are not updated in both the functions. 
I can make the following change, instead of passing as argument to the function, can set it in `Options` of ctx and use it inside ValidateVolume directly, if the option flag of vState is set

Reply:
Moved the flag as part of ScheduleOptions and as a result  there is no need to pass it as parameter to this function